### PR TITLE
Ruby YAML: Add support for plurals

### LINKED ISSUE #3615: Plurals in Ruby YAML Files
As displayed in the good old sample (https://github.com/noosfero/noosfero/blob/master/config/locales/en-US.yml), Ruby (or rails-i18n)'s YAML localization format has some basic plural handling to deal with. This goes down to these `one` `other` things here:

```YML
      less_than_x_minutes:
        one:   "less than a minute"
        other: "less than %{count} minutes"
```

Subkeys used for each language in Ruby's YAML files come directly from [Unicode CIDR Plural Rules](http://www.unicode.org/cldr/charts/28/supplemental/language_plural_rules.html). A basic heuristic for finding plurals is therefore:

1. Look at a key.
2. Does it point to a dictionary? Otherwise return false.
3. Does any of its sub-key members point to a non-string thing? If so, return false.
4. Are all its sub-keys part of the Unicode CIDR cardinal terminology (zero, one, two, few, many, other)? Otherwise return false.
5. Guess true.

Arguably one can never be sure about which key is a plural key, as a perfectly (ab)normal key can also have subkeys named just like plural string groups. Step four can probably improved by some stricter per-language checks.

The current HEAD version does not seem to have such a feature. Since there is no such thing as ruby-yaml-to-po in translate-toolkit, all my observations should be treated as a guess and verified by someone else.

### PR DESCRIPTION
**Pull request by nijel**
Fixes #3615

### CODE REVIEW

File: translate/storage/yaml.py [hunk 1]
```diff
@@ -171,18 +177,22 @@ def nested_set(target, path, value):
             default_flow_style=False, encoding='utf-8', allow_unicode=True
         ))
 
+    def _parse_dict(self, data, prev):
```
Comment (unho) [inline translate/storage/yaml.py]: If I am not wrong getting this block of code out to this new `_parse_dict` method doesn't change the behavior, isn't it? If the answer is yes I would prefer this cleanup to happen in a separate commit.

Comment (unho) [reply translate/storage/yaml.py]: I see now that the intention is to allow to override it on the subclass. Anyway IMHO it is cleaner if this happens in a separate commit.

File: translate/storage/yaml.py [hunk 2]
```diff
@@ -249,3 +259,30 @@ def get_root_node(self, node):
             result[self.targetlanguage] = node
             return result
         return node
+
+    def _parse_dict(self, data, prev):
+        # Does this look like a plural?
+        if all((x in cldr_plural_categories for x in data.keys())):
+            # Ensure we have correct plurals ordering
```
Comment (unho) [inline translate/storage/yaml.py]: Nit: Comments should end with proper punctuation.

File: translate/storage/yaml.py [hunk 3]
```diff
@@ -249,3 +259,30 @@ def get_root_node(self, node):
...
+            values = [data[item] for item in cldr_plural_categories if item in data]
+            yield (prev, multistring(values))
+            return
+
+        # Handle normal dict
+        for ret in super(RubyYAMLFile, self)._parse_dict(data, prev):
```
Comment (unho) [inline translate/storage/yaml.py]: Nit: Use a name that is more explanatory than `ret`.

**Comment by unho** [comment]
Adding this feature would require updating the docs on the YAML format.

**Comment by unho** [comment]
@nijel lgtm once comments are addressed. Thanks.

**Comment by nijel** [comment]
I think I've addressed all of the issues.

**Comment by unho** [comment]
@nijel Thanks. lgtm

**Pull request merged**